pkg/cri/server/events.go

```diff
@@ -325,6 +325,13 @@ func handleContainerExit(ctx context.Context, e *eventtypes.TaskExit, cntr conta
 		}
 	}
 	err = cntr.Status.UpdateSync(func(status containerstore.Status) (containerstore.Status, error) {
+		// Unknown state can only transit to EXITED state, so we need
+		// to handle unknown state here.
+		if status.Unknown {
+			logrus.Debugf("Container %q transited from UNKNOWN to EXITED", cntr.ID)
+			status.Unknown = false
+		}
+
 		// If FinishedAt has been set (e.g. with start failure), keep as
 		// it is.
 		if status.FinishedAt != 0 {
```

Comment:
I like the change but can we go with:
```
if status.FinishedAt == 0 {
  status.Pid = 0
  status.FinishedAt = e.ExitedAt.UnixNano()
  status.ExitCode = int32(e.ExitStatus)
}
if status.Unknown {		
  logrus.Debugf("Container %q transited from UNKNOWN to EXITED", cntr.ID)		
  status.Unknown = false		
}
return status, nil
```

Reply:
good idea, it makes the code more concise